jans-auth-server/client/src/test/java/io/jans/as/client/ws/rs/SetPublicSubjectIdentifierPerClient.java

```diff
@@ -0,0 +1,148 @@
+
+        String idToken = authorizationResponse.getIdToken();
+
+        Jwt jwt = Jwt.parse(idToken);
+        RSAPublicKey publicKey = JwkClient.getRSAPublicKey(
+                jwksUri,
+                jwt.getHeader().getClaimAsString(JwtHeaderName.KEY_ID));
+        RSASigner rsaSigner = new RSASigner(SignatureAlgorithm.RS256, publicKey);
+        assertTrue(rsaSigner.validate(jwt));
+
+        assertEquals(jwt.getClaims().getClaimAsString(JwtClaimName.SUBJECT_IDENTIFIER), expectedSubValue);
+    }
+
+    @Parameters({"redirectUri"})
+    @Test
+    public void setPublicSubjectIdentifierPerClientFail1(final String redirectUri) {
+        showTitle("setPublicSubjectIdentifierPerClientFail1");
+
+        List<ResponseType> responseTypes = Arrays.asList(
+                ResponseType.TOKEN,
+                ResponseType.ID_TOKEN);
+
+        clientRegistration(redirectUri, responseTypes, SubjectType.PAIRWISE, "mail", true);
+    }
+
+    @Parameters({"redirectUri"})
+    @Test
+    public void setPublicSubjectIdentifierPerClientFail2(final String redirectUri) {
+        showTitle("setPublicSubjectIdentifierPerClientFail2");
+
+        List<ResponseType> responseTypes = Arrays.asList(
+                ResponseType.TOKEN,
+                ResponseType.ID_TOKEN);
+
+        clientRegistration(redirectUri, responseTypes, SubjectType.PUBLIC, "invalid_attribute", true);
+    }
+
+    @NotNull
+    private RegisterResponse clientRegistration(
+            String redirectUri, List<ResponseType> responseTypes, SubjectType subjectType,
+            String subjectIdentifierAttribute, boolean checkError) {
+        List<String> redirectUriList = Arrays.asList(redirectUri.split(StringUtils.SPACE));
+        RegisterRequest registerRequest = new RegisterRequest(ApplicationType.WEB, "jans test app", redirectUriList);
+        registerRequest.setSubjectType(subjectType);
+        registerRequest.setResponseTypes(responseTypes);
+        registerRequest.setSubjectIdentifierAttribute(subjectIdentifierAttribute);
+
+        RegisterClient registerClient = new RegisterClient(registrationEndpoint);
+        registerClient.setRequest(registerRequest);
+        RegisterResponse registerResponse = registerClient.exec();
+
+        showClient(registerClient);
+
+        if (checkError) {
+            assertEquals(registerResponse.getStatus(), 400);
+            assertNotNull(registerResponse.getEntity());
+            assertNotNull(registerResponse.getErrorType());
+            assertNotNull(registerResponse.getErrorDescription());
+        } else {
+            assertEquals(registerResponse.getStatus(), 201, "Unexpected response code: " + registerResponse.getEntity());
+            assertNotNull(registerResponse.getClientId());
+            assertNotNull(registerResponse.getClientSecret());
+            assertNotNull(registerResponse.getRegistrationAccessToken());
+            assertNotNull(registerResponse.getClientSecretExpiresAt());
```

Comment:
This falls into `io.jans.as.client.client.Asserter#assertOk`. Can we add same for `assertBadRequest(RegisterResponse response)` ?